packages/ember-routing/lib/system/router.js

```diff
@@ -653,22 +654,94 @@ const EmberRouter = EmberObject.extend(Evented, {
     };
   },
 
-  _serializeQueryParams(targetRouteName, queryParams) {
-    forEachQueryParam(this, targetRouteName, queryParams, function(key, value, qp) {
-      delete queryParams[key];
-      queryParams[qp.urlKey] = qp.route.serializeQueryParam(value, qp.urlKey, qp.type);
+  /**
+    Serializes the given query params according to their QP meta information.
+
+    @private
+    @method _serializeQueryParams
+    @param {TransitionState} state
+    @param {Object} queryParams
+    @return {Void}
+  */
+  _serializeQueryParams(state, queryParams) {
+    forEachQueryParam(this, state, queryParams, (key, value, qp) => {
+      if (qp) {
+        delete queryParams[key];
+        queryParams[qp.urlKey] = qp.route.serializeQueryParam(value, qp.urlKey, qp.type);
+      } else {
+        queryParams[key] = this._serializeQueryParam(value, typeOf(value));
+      }
     });
   },
 
-  _deserializeQueryParams(targetRouteName, queryParams) {
-    forEachQueryParam(this, targetRouteName, queryParams, function(key, value, qp) {
-      delete queryParams[key];
-      queryParams[qp.prop] = qp.route.deserializeQueryParam(value, qp.urlKey, qp.type);
+  /**
+    Serializes value of a query parameter based on a type
+
+    @private
+    @method _serializeQueryParam
+    @param {Object} value
+    @param {String} type
+  */
+  _serializeQueryParam(value, type) {
+    if (type === 'array') {
+      return JSON.stringify(value);
+    }
+
+    return `${value}`;
```

Comment:
This was copied form the original function in route.js. That said, I actually like this as a way of coercing a string value.